include/mesh.h

```diff
@@ -0,0 +1,273 @@
+#ifndef PIPE_NETWORK_MESH_H_
```

Comment:
Header should only declare the functions, not define them. Use a source file to define your functions.